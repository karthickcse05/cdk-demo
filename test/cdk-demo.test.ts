import { Template}from '@aws-cdk/assertions';
import * as cdk from '@aws-cdk/core';
import * as cdkdemo from '../lib/cdk-demo-stack'
import { SynthUtils} from '@aws-cdk/assert';

// unit testing 
const app = new cdk.App();
// fine grained

test('test to check the queue name',()=>{

const stack = new cdkdemo.CdkDemoStack(app,'Myteststack');

const template = Template.fromStack(stack);

template.hasResourceProperties('AWS::SQS::Queue',{
    QueueName: 'demo-jan29'
});

});


// snapshot

test('test app for snapshot',()=>{
    const stack = new cdkdemo.CdkDemoStack(app,'snapshottest');
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();

});