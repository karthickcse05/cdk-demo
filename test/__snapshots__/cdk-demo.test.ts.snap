// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test app for snapshot 1`] = `
Object {
  "Resources": Object {
    "CdkDemoQueue32E7553B": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "QueueName": "demo-jan29",
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;
